### PR TITLE
the rest of plumbing for FS notify

### PR DESCRIPTION
**Pull request by maxtaco**
the rest of the plumbing work for FS notify

### CODE REVIEW

**Comment by maxtaco** [comment]
r? @strib 

Follow on to #4750

File: go/libkb/notify_router.go [hunk 1]
```diff
@@ -337,6 +339,54 @@ func (n *NotifyRouter) HandleFSEditListRequest(ctx context.Context, arg keybase1
...
+	if n == nil {
+		return
+	}
+	// For all connections we currently have open...
+	n.cm.ApplyAll(func(id ConnectionID, xp rpc.Transporter) bool {
+		// If the connection wants the `Kbfsrequest` notification type
+		if n.getNotificationChannels(id).Kbfsrequest {
+			// In the background do...
+			go func() {
+				// A send of a `FSEditListRequest` RPC with the notification
```
Comment (strib) [inline go/libkb/notify_router.go]: Wrong comment.

File: go/libkb/notify_router.go [hunk 2]
```diff
@@ -337,6 +339,54 @@ func (n *NotifyRouter) HandleFSEditListRequest(ctx context.Context, arg keybase1
...
```
Comment (strib) [inline go/libkb/notify_router.go]: Wrong comment here too.

File: go/libkb/notify_router.go [hunk 3]
```diff
@@ -337,6 +339,54 @@ func (n *NotifyRouter) HandleFSEditListRequest(ctx context.Context, arg keybase1
...
 
+// HandleFSSyncStatus is called for KBFS sync status notifications.
+func (n *NotifyRouter) HandleFSSyncStatus(ctx context.Context, arg keybase1.FSSyncStatusArg) {
+	if n == nil {
+		return
+	}
+	// For all connections we currently have open...
+	n.cm.ApplyAll(func(id ConnectionID, xp rpc.Transporter) bool {
+		// If the connection wants the `Kbfsrequest` notification type
+		if n.getNotificationChannels(id).Kbfsrequest {
```
Comment (strib) [inline go/libkb/notify_router.go]: Should be `Kbfs`, not `Kbfsrequest` (in the comment too).

File: go/libkb/notify_router.go [hunk 4]
```diff
@@ -337,6 +339,54 @@ func (n *NotifyRouter) HandleFSEditListRequest(ctx context.Context, arg keybase1
...
+
+// HandleFSSyncEvent is called for KBFS sync event notifications.
+func (n *NotifyRouter) HandleFSSyncEvent(ctx context.Context, arg keybase1.FSPathSyncStatus) {
+	if n == nil {
+		return
+	}
+	// For all connections we currently have open...
+	n.cm.ApplyAll(func(id ConnectionID, xp rpc.Transporter) bool {
+		// If the connection wants the `Kbfsrequest` notification type
+		if n.getNotificationChannels(id).Kbfsrequest {
```
Comment (strib) [inline go/libkb/notify_router.go]: Same here.

**Comment by maxtaco** [comment]
Sorry for being so sloppy here...

**Comment by maxtaco** [comment]
Ok, problems addressed.  Found some related issues too, for a subsequent PR...

**Comment by maxtaco** [comment]
(See #4765 for other issues that came up)

**Comment by maxtaco** [comment]
C/I error is an unfortunately recurring flake

**Comment by maxtaco** [comment]
@strib  is this ok?

**Comment by maxtaco** [comment]
'tis

**Pull request merged**